### ISSUE TITLE
MyPy should receive the python interpreter used in the project

### ISSUE THREAD

**Issue by jmfederico** [microsoft/vscode-mypy#191]
## Environment data

-   VS Code version: 1.44.2
-   Extension version (available under the Extensions sidebar): 2020.4.76186
-   OS and version: macOS Catalina 10.15.4 (19E287)
-   Python version: 3.8.2
-   Type of virtual environment used: venv
-   Relevant/affected Python packages and their versions: mypy 0.770
-   Jedi or Language Server?: Jedi
-   Value of the `python.languageServer` setting: Microsoft

## Expected behaviour

When running mypy, it should receive `--python-executable` with the path to the active python interpreter if one is set.

## Actual behaviour

When mypy runs, it lints with Python interpreter available to it, and not the one selected as interpreter in VSCode, and since `--ignore-missing-imports` is passed to mypy by default, it can be common that types are not correctly resolved, and errors are just being muted.

## Steps to reproduce:
1. Install MyPy globally (or with pipx)
2. Create a virtual environment and open the project with VSCode
    ```python3 -m venv .venv```
3. Enable MyPy linting `"python.linting.mypyEnabled": true` and `"python.linting.mypyPath": "/path/to/bin/mypy"`
4. Set the Python interpreter to the one in the virtual environment `"python.pythonPath": ".venv/bin/python"`
5. Install some library that has custom types (I use boto3-stubs) `.venv/bin/pip install boto3-stubs`
6. Create a file with:
    ```python
    from mypy_boto3 import boto3_session
    session: boto3_session.Session = 123
    ```
6. MyPy will not report any errors

Now, to fix that problem:
1. Add `--python-executable`  to the list of arguments passed to MyPy:
    ```json
    "python.linting.mypyArgs": [
      "--ignore-missing-imports",
      "--follow-imports=silent",
      "--show-column-numbers",
      "--python-executable=.venv/bin/python3"
    ]
    ```
2. MyPy will now report a type error: `Incompatible types in assignment (expression has type "int", variable has type "Session")`

I set this as a bug and not a feature request because VSCode knows what python interpreter the user wants to use, as it is a setting that the user can change. It would make sense that the interpreter is used when it affects the behaviour of other tools.
